### DISCUSSION TITLE
RxJS 8 support

### DISCUSSION THREAD

**Discussion by e-oz** [ngrx/platform#3365] [General]
I would like to know contributors' opinions about how difficult it would be to replace usage of rxjs/Notification in @ngrx/store library (Notification was removed in RxJS 8).

**Comment by timdeschryver** [comment]
I think most of the steps are already taken, which is using the const `N`, `E`, and `C` instead of `Notification`. Why the question? Do you spot something that could be difficult here?

**Reply by e-oz** [reply]
Only because current @ngrx/store is incompatible with rxjs/8-alpha. If it's handled already in some separate branch - good to know!

**Reply by timdeschryver** [reply]
To be clear, there's currently not a branch that updates RxJS to v8.

**Comment by e-oz** [comment]
So, affected files are in `modules/effects/src/`:
[act.ts](https://github.com/ngrx/platform/blob/master/modules/effects/src/act.ts)
[effect_notification.ts](https://github.com/ngrx/platform/blob/master/modules/effects/src/effect_notification.ts)
[effect_sources.ts](https://github.com/ngrx/platform/blob/master/modules/effects/src/effect_sources.ts)

They import `Notification` class from "rxjs".
This class was [deprecated in RxJS 7](https://rxjs.dev/api/index/class/Notification) and removed from 8.

I suppose, we could create an interface or a class "Notification" in `modules/effects/src/`, something like
```ts
export interface Notification<T> {
  hasValue: boolean;
  kind: 'N' | 'E' | 'C';
  value?: T;
  error?: any;
}
```
and just edit the imports in 3 files.

If you think that it's ok - I'm ready to create a PR. If you have a better idea or I'm missing some important details - please let me know.

**Reply by timdeschryver** [reply]
Because it's removed I'm not sure if we should create our own.
Instead, we could use the [ObservableNotification](https://github.com/ReactiveX/rxjs/blob/8.x/src/internal/types.ts#L140) interface from RxJS.

**Reply by e-oz** [reply]
By your suggestion: PR https://github.com/ngrx/platform/pull/3407
